src/main/java/org/shanerx/tradeshop/utils/configuration/interfaces/YAMLConfigurationSectionInterface.java

```diff
@@ -0,0 +1,48 @@
+/*
+ *
+ *                         Copyright (c) 2016-2019
+ *                SparklingComet @ http://shanerx.org
+ *               KillerOfPie @ http://killerofpie.github.io
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *                http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ *
+ *  NOTICE: All modifications made by others to the source code belong
+ *  to the respective contributor. No contributor should be held liable for
+ *  any damages of any kind, whether be material or moral, which were
+ *  caused by their contribution(s) to the project. See the full License for more information.
+ *
+ */
+
+package org.shanerx.tradeshop.utils.configuration.interfaces;
+
+public interface YAMLConfigurationSectionInterface {
```

Comment:
This should be a subinterface of YAMLConfigurationInterface (same concept as ConfigurationSection being more or less a subclass of YamlConfiguration), as many methods are the same.

Reply:
Many are the same but YAMLConfigurationInterface has 

    Object getDefaultValue();

    YAMLConfigurationSectionInterface getSectionKey();

DefaultValues arn't in config sections in any way and I don't want the method to be required there and getSectionKey is changed to getParent because its more specific and not required there(although this can just be renamed here)

Either way I think DefaultValue prevents me from doing this. which ConfigurationSection and YamlConfiguration both dont have

Reply:
Just make it return null if it is a config section, or an throw an exception. It shouldn't come to that anyway.

Reply:
I would still have to add the method into the config section which would be even more wasteful then repeating the couple similar methods here since it theoretically would never be called. I think at least. Either way this isn't in the new implementation so unless I end up going back it probably wont matter.